### PR TITLE
Update LUCI APIs to query luci builders by using bucket and repo

### PR DESCRIPTION
**Pull request by keyonghan**
This is part of flutter/flutter#62429

### CODE REVIEW

File: app_dart/lib/src/service/luci.dart [hunk 1]
```diff
@@ -194,8 +194,8 @@ class LuciBuilder {
   Map<String, dynamic> toJson() => _$LuciBuilderToJson(this);
 
   /// Loads and returns the list of known builders from the Cocoon [config].
-  static Future<List<LuciBuilder>> getBuilders(Config config) async {
-    final List<dynamic> builders = await config.luciProdBuilders;
+  static Future<List<LuciBuilder>> getBuilders(String repo, Config config) async {
```
Comment (godofredoc) [inline app_dart/lib/src/service/luci.dart:197]: Should this be named as getProdBuilders or pass the bucket as parameter?

Comment (keyonghan) [reply app_dart/lib/src/service/luci.dart]: Changed to `getProdBuilders`.

File: app_dart/test/request_handlers/refresh_chromebot_status_test.dart [hunk 2]
```diff
@@ -109,9 +103,13 @@ void main() {
       final Task task = Task(key: commit.key.append(Task, id: 123), status: Task.statusNew);
       config.db.values[commit.key] = commit;
       config.db.values[task.key] = task;
+      //config.flutterLuciProdBuildersValue = <Map<String, dynamic>>[
+      //  <String, String>{'name': 'test', 'repo': 'flutter', 'TaskName': 'test'}
+      //];
```
Comment (digiter) [inline app_dart/test/request_handlers/refresh_chromebot_status_test.dart:108]: rm

Comment (keyonghan) [reply app_dart/test/request_handlers/refresh_chromebot_status_test.dart]: Done.

**Pull request merged**